src/orca/errors.py

```diff
@@ -11,3 +11,11 @@ class ClientRequestError(OrcaError):
 
 class ClientArgsError(OrcaError):
     """Client arguments are missing or invalid."""
+
+
+class OptionalAttrRequiredError(OrcaError):
```

Comment:
Nice usage of custom errors here. Just for @BWMac , we always want to use the python built in exceptions as much as possible first prior to building our own.